kms/api-client/functions.py

```diff
@@ -0,0 +1,292 @@
+    call to decrypt."""
+
+    # Creates an API client for the KMS API.
+    kms_client = discovery.build('cloudkms', 'v1beta1')
+
+    # The resource name of the CryptoKey.
+    name = 'projects/{}/locations/{}/keyRings/{}/cryptoKeys/{}'.format(
+        project_id, location, keyring, cryptokey)
+
+    # Read text from the input file.
+    with io.open(plaintext_file_name, 'rb') as plaintext_file:
+        plaintext = plaintext_file.read()
+        encoded_text = base64.b64encode(plaintext)
+
+        # Use the KMS API to encrypt the text.
+        cryptokeys = kms_client.projects().locations().keyRings().cryptoKeys()
+        request = cryptokeys.encrypt(
+            name=name, body={"plaintext": encoded_text.decode('utf-8')})
+        response = request.execute()
+
+        # Write the encrypted text to a file.
+        with io.open(encrypted_file_name, 'wb') as encrypted_file:
+            encrypted_file.write(response["ciphertext"].encode('utf-8'))
+
+    print('Saved encrypted text to {}.'.format(encrypted_file_name))
+# [END kms_encrypt]
+
+
+# [START kms_decrypt]
+def decrypt(project_id, location, keyring, cryptokey, encrypted_file_name,
+            decrypted_file_name):
+    """Decrypts data from encrypted_file_name that was previously encrypted
+    using the CryptoKey with a call to encrypt. Outputs decrypted data to
+    decrpyted_file_name."""
+
+    # Creates an API client for the KMS API.
+    kms_client = discovery.build('cloudkms', 'v1beta1')
+
+    # The resource name of the CryptoKey.
+    name = 'projects/{}/locations/{}/keyRings/{}/cryptoKeys/{}'.format(
+        project_id, location, keyring, cryptokey)
+
+    # Read cipher text from the input file.
+    with io.open(encrypted_file_name, 'rb') as encrypted_file:
+        cipher_text = encrypted_file.read()
+
+        # Use the KMS API to decrypt the text.
+        cryptokeys = kms_client.projects().locations().keyRings().cryptoKeys()
+        request = cryptokeys.decrypt(
+            name=name, body={"ciphertext": cipher_text.decode('utf-8')})
+        response = request.execute()
+
+        # Write the plain text to a file.
+        with io.open(decrypted_file_name, 'wb') as decrypted_file:
+            plaintext_encoded = response["plaintext"]
+            plaintext_decoded = base64.b64decode(plaintext_encoded)
+            decrypted_file.write(plaintext_decoded)
+
+    print('Saved decrypted text to {}.'.format(decrypted_file_name))
+# [END kms_decrypt]
+
+
+# [START kms_disable_cryptokey_version]
+def disable_cryptokey_version(project_id, location, keyring, cryptokey,
+                              version):
+    """Disables a CryptoKeyVersion associated with a given CryptoKey and
+    KeyRing."""
+
+    # Creates an API client for the KMS API.
+    kms_client = discovery.build('cloudkms', 'v1beta1')
+
+    # Construct the resource name of the CryptoKeyVersion.
+    name = 'projects/{}/locations/{}/'.format(project_id, location)
```

Comment:
do this to break the string up over multiple lines:

```python
name = (
    'projects/..........'
    '/....'
    '.....')
```

Python will automatically concatenate them.